### ISSUE TITLE
ValueError is raised with custom reaction identifiers

### ISSUE THREAD

**Issue by mostueve** [micom-dev/micom#8]
When trying to build a community, I always get a ValueError similar to the one below.

ValueError: ('The model already contains a reaction with the id:', 'ABC_RXN__smaAA1')

Files causing this can be found at https://gitlab.com/mostueve/micom-error-files

**Comment by cdiener** [comment]
Hi, thanks for providing a reproducible example. That helped a lot. Micom makes some adjustments to the IDs and that did not work for the models you provided. I fixed that in version 0.9.13, and updating to that version will make your setup work. You will see some SBML errors and the micom logs since your models are invalid SBML (missing a model ID), but it will read anyways. Those message will disappear if you set a model id in your SBML. 

Anyways it will build the community and can obtain solutions.

## Test run

<details>

```python
In [1]: import micom

In [2]: import pandas as pd

In [3]: tax = pd.read_csv("pandas_metacyc.csv", sep="\t")

In [4]: com = micom.Community(tax)
  0%|                                                                                                                                                                                     | 0/4 [00:00<?, ?models/s]ERROR:root:'' is not a valid SBML 'SId'.
INFO:micom:converting IDs for smaAA1
INFO:micom:adding reactions for smaAA1 to community
INFO:micom:adding metabolite GLC_m to external medium
INFO:micom:adding metabolite WATER_m to external medium
INFO:micom:adding metabolite PROTON_m to external medium
INFO:micom:adding metabolite OXYGEN__MOLECULE_m to external medium
INFO:micom:adding metabolite Pi_m to external medium
INFO:micom:adding metabolite MN2_m to external medium
INFO:micom:adding metabolite ZN2_m to external medium
INFO:micom:adding metabolite CU2_m to external medium
INFO:micom:adding metabolite CA2_m to external medium
INFO:micom:adding metabolite SULFATE_m to external medium
INFO:micom:adding metabolite AMMONIA_m to external medium
INFO:micom:adding metabolite ADP_cof_m to external medium
INFO:micom:adding metabolite GDP_cof_m to external medium
INFO:micom:adding metabolite NAD_cof_m to external medium
INFO:micom:adding metabolite NADP_cof_m to external medium
INFO:micom:adding metabolite THF_cof_m to external medium
INFO:micom:adding metabolite CO__A_cof_m to external medium
INFO:micom:adding metabolite Acceptor_cof_m to external medium
INFO:micom:adding metabolite Oxidized__ferredoxins_cof_m to external medium
INFO:micom:adding metabolite Ox__NADPH__Hemoprotein__Reductases_cof_m to external medium
INFO:micom:adding metabolite Cytochromes__C__Oxidized_cof_m to external medium
INFO:micom:adding metabolite Oxidized__Plastocyanins_cof_m to external medium
 25%|███████████████████████████████████████████▎                                                                                                                                 | 1/4 [00:08<00:26,  8.67s/models]INFO:micom:reading model from ecl_com.xml
ERROR:root:'' is not a valid SBML 'SId'.
INFO:micom:converting IDs for eclAA4
INFO:micom:adding reactions for eclAA4 to community
INFO:micom:updating import rate for external metabolite GLC_m
INFO:micom:updating import rate for external metabolite WATER_m
INFO:micom:updating import rate for external metabolite PROTON_m
INFO:micom:updating import rate for external metabolite OXYGEN__MOLECULE_m
INFO:micom:updating import rate for external metabolite Pi_m
INFO:micom:updating import rate for external metabolite MN2_m
INFO:micom:updating import rate for external metabolite ZN2_m
INFO:micom:updating import rate for external metabolite CU2_m
INFO:micom:updating import rate for external metabolite CA2_m
INFO:micom:updating import rate for external metabolite SULFATE_m
INFO:micom:updating import rate for external metabolite AMMONIA_m
INFO:micom:updating import rate for external metabolite ADP_cof_m
INFO:micom:updating import rate for external metabolite GDP_cof_m
INFO:micom:updating import rate for external metabolite NAD_cof_m
INFO:micom:updating import rate for external metabolite NADP_cof_m
INFO:micom:updating import rate for external metabolite THF_cof_m
INFO:micom:updating import rate for external metabolite CO__A_cof_m
INFO:micom:updating import rate for external metabolite Acceptor_cof_m
INFO:micom:updating import rate for external metabolite Oxidized__ferredoxins_cof_m
INFO:micom:updating import rate for external metabolite Ox__NADPH__Hemoprotein__Reductases_cof_m
INFO:micom:updating import rate for external metabolite Cytochromes__C__Oxidized_cof_m
INFO:micom:updating import rate for external metabolite Oxidized__Plastocyanins_cof_m
 50%|██████████████████████████████████████████████████████████████████████████████████████▌                                                                                      | 2/4 [00:20<00:19,  9.66s/models]INFO:micom:reading model from ppu_com.xml
ERROR:root:'' is not a valid SBML 'SId'.
INFO:micom:converting IDs for ppuAA7
INFO:micom:adding reactions for ppuAA7 to community
INFO:micom:updating import rate for external metabolite GLC_m
INFO:micom:updating import rate for external metabolite WATER_m
INFO:micom:updating import rate for external metabolite PROTON_m
INFO:micom:updating import rate for external metabolite OXYGEN__MOLECULE_m
INFO:micom:updating import rate for external metabolite Pi_m
INFO:micom:updating import rate for external metabolite MN2_m
INFO:micom:updating import rate for external metabolite ZN2_m
INFO:micom:updating import rate for external metabolite CU2_m
INFO:micom:updating import rate for external metabolite CA2_m
INFO:micom:updating import rate for external metabolite SULFATE_m
INFO:micom:updating import rate for external metabolite AMMONIA_m
INFO:micom:updating import rate for external metabolite ADP_cof_m
INFO:micom:updating import rate for external metabolite GDP_cof_m
INFO:micom:updating import rate for external metabolite NAD_cof_m
INFO:micom:updating import rate for external metabolite NADP_cof_m
INFO:micom:updating import rate for external metabolite THF_cof_m
INFO:micom:updating import rate for external metabolite CO__A_cof_m
INFO:micom:updating import rate for external metabolite Acceptor_cof_m
INFO:micom:updating import rate for external metabolite Oxidized__ferredoxins_cof_m
INFO:micom:updating import rate for external metabolite Ox__NADPH__Hemoprotein__Reductases_cof_m
INFO:micom:updating import rate for external metabolite Cytochromes__C__Oxidized_cof_m
INFO:micom:updating import rate for external metabolite Oxidized__Plastocyanins_cof_m
 75%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                                           | 3/4 [00:32<00:10, 10.27s/models]INFO:micom:reading model from hfr_com.xml
ERROR:root:'' is not a valid SBML 'SId'.
INFO:micom:converting IDs for hfrAA6
INFO:micom:adding reactions for hfrAA6 to community
INFO:micom:updating import rate for external metabolite GLC_m
INFO:micom:updating import rate for external metabolite WATER_m
INFO:micom:updating import rate for external metabolite PROTON_m
INFO:micom:updating import rate for external metabolite OXYGEN__MOLECULE_m
INFO:micom:updating import rate for external metabolite Pi_m
INFO:micom:updating import rate for external metabolite MN2_m
INFO:micom:updating import rate for external metabolite ZN2_m
INFO:micom:updating import rate for external metabolite CU2_m
INFO:micom:updating import rate for external metabolite CA2_m
INFO:micom:updating import rate for external metabolite SULFATE_m
INFO:micom:updating import rate for external metabolite AMMONIA_m
INFO:micom:updating import rate for external metabolite ADP_cof_m
INFO:micom:updating import rate for external metabolite GDP_cof_m
INFO:micom:updating import rate for external metabolite NAD_cof_m
INFO:micom:updating import rate for external metabolite NADP_cof_m
INFO:micom:updating import rate for external metabolite THF_cof_m
INFO:micom:updating import rate for external metabolite CO__A_cof_m
INFO:micom:updating import rate for external metabolite Acceptor_cof_m
INFO:micom:updating import rate for external metabolite Oxidized__ferredoxins_cof_m
INFO:micom:updating import rate for external metabolite Ox__NADPH__Hemoprotein__Reductases_cof_m
INFO:micom:updating import rate for external metabolite Cytochromes__C__Oxidized_cof_m
INFO:micom:updating import rate for external metabolite Oxidized__Plastocyanins_cof_m
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:45<00:00, 11.18s/models]

In [5]: sol = com.cooperative_tradeoff(fraction=0.5)
INFO:micom:setting growth rate constraint for smaAA1
INFO:micom:minimal growth rate smaller than tolerance, setting to zero.
INFO:micom:setting growth rate constraint for eclAA4
INFO:micom:minimal growth rate smaller than tolerance, setting to zero.
INFO:micom:setting growth rate constraint for ppuAA7
INFO:micom:minimal growth rate smaller than tolerance, setting to zero.
INFO:micom:setting growth rate constraint for hfrAA6
INFO:micom:minimal growth rate smaller than tolerance, setting to zero.
INFO:micom:adding L2 norm to None
INFO:micom:finished adding tradeoff objective to None
INFO:micom:solver returned the status numeric, returning the solution anyway.
INFO:micom:Starting crossover...
INFO:micom:constraining growth rates.
INFO:micom:finding closest feasible solution
INFO:micom:resetting growth rate constraint for hfrAA6
INFO:micom:resetting growth rate constraint for ppuAA7
INFO:micom:resetting growth rate constraint for eclAA4
INFO:micom:resetting growth rate constraint for smaAA1

In [6]: sol
Out[6]: <CommunitySolution 5.152 at 0x129b2e9b0>

In [7]: sol.members
Out[7]:
              abundance  growth_rate  reactions  metabolites
compartments
eclAA4              0.6     7.359998       3246         1531
hfrAA6              0.2     2.453333       3411         1538
medium              NaN          NaN         22           22
ppuAA7              0.1     1.226668       3051         1493
smaAA1              0.1     1.226667       2626         1262
```

</details>

**Comment by mostueve** [comment]
Cool, thanks a lot! Now it is working.
